packages/pwa/app/utils/routes-utils.js

```diff
@@ -7,6 +7,27 @@
 
 import {getSites} from './site-utils'
 import {urlPartPositions} from '../constants'
+import {matchPath} from 'react-router-dom'
+
+/**
+ * Check if a path matches an array of routes.
+ * @param path path that
+ * @param routes
+ * @returns {*}
+ */
+export const matchRoute = (path, routes) => {
```

Comment:
You might also consider returning the matched route here which would allow you to decide whether to discard the matched route (`*`).

Reply:
Good suggestion! that way we'll not need to remove the `*` path from the `routes` array.